src/utils/wallet.js

```diff
@@ -55,17 +55,21 @@ const wallet = {
    * @param {string} passphrase
    * @param {string} pin
    * @param {string} password
-   * @param {boolean} loadHistory if should load history from generated addresses
    *
-   * @return {string} words generated (null if words are not valid)
    * @memberof Wallet
    * @inner
    */
-  generateWallet(words, passphrase, pin, password, loadHistory) {
-    if (hathorLib.wallet.wordsValid(words).valid) {
-      return this.executeGenerateWallet(words, passphrase, pin, password, loadHistory);
-    } else {
-      return null;
+  generateWallet(words, passphrase, pin, password, routerHistory) {
+    try {
+      walletUtils.wordsValid(words);
+      return this.startWallet(words, passphrase, pin, password, routerHistory);
```

Comment:
Maybe we should move this out of the try/except block. As far as I understood, we not handling any exceptions that might be caused by it, right?

Reply:
Yes, you're right. Moved out.